pyproject.toml

```diff
@@ -48,6 +49,197 @@ Code = "https://github.com/SciTools/nc-time-axis"
+
+    # -> No examples section found
+    "EX01",  # Not all docstrings require an "Examples" section.
+
+    # -> No Yields section found
+    "YD01",  # Not all docstrings require a "Yields" section.
+
+    # TODO: exceptions that still need investigating are below.
+    # Might be fixable, or might become permanent (above):
+    "GL08",  # No docstring
+    "SS05",  # Summary must start with infinitive verb
+    "SS06",  # Summary should fit on one line
+    "PR01",  # Parameters not documented
+    "PR06",  # Wrong type used
+    "PR08",  # Description should start with capitol letter
+    "RT01",  # No Returns section found
+]
+exclude = [
+    '\.__eq__$',
+    '\.__ne__$',
+    '\.__repr__$',
+]
+
+[tool.pytest.ini_options]
+addopts = "-ra -v --doctest-modules --strict-config --strict-markers"
+doctest_optionflags = "NORMALIZE_WHITESPACE ELLIPSIS NUMBER"
+#filterwarnings = ["error"] # TODO - PP309; enable once all warnings are fixed
+log_cli_level = "INFO"
+minversion = "6.0"
+testpaths = ["src/nc_time_axis"]
+xfail_strict = true
+
+[tool.repo-review]
+ignore = [
+    # https://learn.scientific-python.org/development/guides/style/#PC180
+    "PC180",  # Uses prettier
+
+    # TODO: exceptions that still need investigating are below.
+    # Might be fixable, or might become permanent (above):
+    "PY007", # Supports an easy task runner (nox or tox)
+    "PP309", # Filter warnings specified
+    "PC170", # Uses PyGrep hooks (only needed if rST present)
+]
+
+[tool.ruff]
+line-length = 88
+
+[tool.ruff.format]
+preview = false
+
+[tool.ruff.lint]
+ignore = [
+    # flake8-commas (COM)
+    # https://docs.astral.sh/ruff/rules/#flake8-commas-com
+    "COM812",  # Trailing comma missing.
+    "COM819",  # Trailing comma prohibited.
+
+    # flake8-implicit-str-concat (ISC)
+    # https://docs.astral.sh/ruff/rules/single-line-implicit-string-concatenation/
+    # NOTE: This rule may cause conflicts when used with "ruff format".
+    "ISC001",  # Implicitly concatenate string literals on one line.
+
+    # TODO: exceptions that still need investigating are below.
+    # Might be fixable, or might become permanent (above):
+    "PLR5501", # collapsible-else-if; Preserve readability of TODO block in
+               # `convert` method
+
+    "A001",
+    "A002",
+    "ANN001",
+    "ANN002",
+    "ANN003",
+    "ANN201",
+    "ANN202",
+    "ANN204",
+    "ANN205",
+    "ANN206",
+    "ARG001",
+    "ARG002",
+    "ARG003",
+    "ARG004",
+    "B028",
+    "C408",
+    "C901",
+    "D100",
+    "D101",
+    "D102",
+    "D103",
+    "D105",
+    "D200",
+    "D205",
+    "D401",
+    "DTZ002",
+    "E501",
+    "EM101",
+    "ERA001",
+    "FIX002",
+    "ICN001",
+    "N801",
+    "N802",
+    "NPY002",
+    "PLR0912",
+    "PT009",
+    "PT027",
+    "RET503",
+    "S101",
+    "SIM102",
+    "SIM108",
+    "TD002",
+    "TD003",
+    "TD004",
+    "TRY003",
+    "TRY004",
+    "UP008",
```

Comment:
I think it's worth annotating these with the error message these give (or a paraphrase of the error message less specific to the actual error). For iris-esmf-regrid, I grouped together similar errors such as `ANN***`.

Reply:
Done; see 9e31436
I have also updated the associated Issue #297